packages/core/helper-plugin/src/features/RBAC.tsx

```diff
@@ -1,36 +1,25 @@
 import * as React from 'react';
 
-/**
- * @preserve
- * @typedef {Object} Permission
- * @property {string} action
- * @property {unknown[]} conditions
- * @property {number} id
- * @property {Record<string, unknown>} properties
- * @property {string} subject
- */
+interface Permission {
+  action: string;
+  conditions: unknown[];
+  id: number;
+  properties: Record<string, unknown>;
+  subject: string | null;
+}
 
-/**
- * @preserve
- * @typedef {import('react-query').QueryObserverBaseResult<Permission[]>['refetch']} RefetchPermissionsFn
- */
+type RefetchPermissionsFn = import('react-query').QueryObserverBaseResult<Permission[]>['refetch'];
 
 /* -------------------------------------------------------------------------------------------------
  * Context
  * -----------------------------------------------------------------------------------------------*/
 
-/**
- * @preserve
- * @typedef {Object} RBACContextValue
- * @property {Permission[]} allPermissions – The permissions of the current user.
- * @property {RefetchPermissionsFn} refetchPermissions
- */
+type RBACContextValue = {
+  allPermissions: Permission[]; // The permissions of the current user.
+  refetchPermissions: RefetchPermissionsFn;
+};
```

Comment:
I would make these optional because then you don't have to cast the below, because your cast is technically incorrect – if we don't add them to the context then they don't exist in the value 👀, you could alternatively set the default to null, so it's either an object with definitely those properties, or null... what do you think works best?